code/game/machinery/telecomms/broadcasting.dm

```diff
@@ -53,127 +53,141 @@
 
 **/
 
+// Subtype of /datum/signal with additional processing information.
+/datum/signal/subspace
+	transmission_method = TRANSMISSION_SUBSPACE
+	var/server_type = /obj/machinery/telecomms/server
+	var/datum/signal/subspace/original
+	var/list/levels
+
+/datum/signal/subspace/New(data)
+	src.data = data || list()
+
+/datum/signal/subspace/proc/copy()
+	var/datum/signal/subspace/copy = new
+	copy.original = src
+	copy.source = source
+	copy.levels = levels
+	copy.frequency = frequency
+	copy.server_type = server_type
+	copy.transmission_method = transmission_method
+	copy.data = data.Copy()
+	return copy
+
+/datum/signal/subspace/proc/mark_done()
+	var/datum/signal/subspace/current = src
+	while (current)
+		current.data["done"] = TRUE
+		current = current.original
+
+/datum/signal/subspace/proc/send_to_receivers()
+	for(var/obj/machinery/telecomms/receiver/R in GLOB.telecomms_list)
+		R.receive_signal(src)
+	for(var/obj/machinery/telecomms/allinone/R in GLOB.telecomms_list)
+		R.receive_signal(src)
 
-/proc/Broadcast_Message(var/atom/movable/AM,
-						var/vmask, var/obj/item/device/radio/radio,
-						var/message, var/name, var/job, var/realname,
-						var/data, var/compression, var/list/level, var/freq, var/list/spans,
-						var/verb_say, var/verb_ask, var/verb_exclaim, var/verb_yell, var/datum/language/language)
-
+/datum/signal/subspace/proc/broadcast()
 	set waitfor = FALSE
 
-	message = copytext(message, 1, MAX_BROADCAST_LEN)
+// Vocal transmissions (i.e. using saycode).
+// Despite "subspace" in the name, these transmissions can also be RADIO
```

Comment:
Why the need for the /subpsace subtype here then ? Seems like modifying the basic datum would make more sense ?
As in add the common functionality to basic signal then just have signal/vocal ? original message/server and levels seem pretty generic properties to me.